### PR TITLE
Build the first, barebones version

### PR DESCRIPTION
**Pull request by miguelcardo**
This version will get the UUID as user input before calling the SEC Client via the TRANSCEIVE intent

### CODE REVIEW

**Comment by miguelcardo** [comment]
Tested with microSD and contactless cards. Works fine.
Used Android 4.2.2 because it allows choosing a layout - the resulting GUI is both easier and looks much better.

Ideas for the future: create a second tab where the command and response APDUs are logged. Currently the feedback is very poor.

**Comment by miguelcardo** [comment]
@petter-fidesmo Ready for review!

File: src/com/fidesmo/sec/transceivedemo/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		transceiveToCardButton.setOnClickListener(new OnClickListener() {
+			@Override
+			public void onClick(View v) {
+				Log.i(TAG, "Pushed the Transceive To Card button");
+				String uuid = uuidInput.getText().toString();
+				if (checkInputUuid(uuid)) {
+					// create Intent to call the SEC Client's activity to send APDUs to a contactless card
+					// through the phone's NFC interface
+					Intent intent = new Intent(transceiveCardAction, Uri.parse(intentUri + uuid));
+			        startActivity(intent);
```
Comment (ghost) [inline src/com/fidesmo/sec/transceivedemo/MainActivity.java]: This line is not correctly indented.

File: src/com/fidesmo/sec/transceivedemo/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+			@Override
+			public void onClick(View v) {
+				Log.i(TAG, "Pushed the Transceive To Device Fidelity button");
+				String uuid = uuidInput.getText().toString();
+				if (checkInputUuid(uuid)) {
+					// create Intent to call the SEC Client's activity to send APDUs to a DeviceFidelity
+					// microSD card
+					Intent intent = new Intent(transceiveMicroSDAction, Uri.parse(intentUri + uuid));
+			        startActivity(intent);
+				}
```
Comment (ghost) [inline src/com/fidesmo/sec/transceivedemo/MainActivity.java]: This contradicts Java style. It should be `} else {` please refer to http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-142311.html#449

File: src/com/fidesmo/sec/transceivedemo/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+				else {
+					// ask user to enter a valid UUID
+					wrongUuid();
+				}		
+			}
+		});
+		
+	}
+	
+	boolean checkInputUuid(String input) {
```
Comment (ghost) [inline src/com/fidesmo/sec/transceivedemo/MainActivity.java]: This is a little bit of a boilerplate construct, a more common Java style is to replace the "if else" in the calling code with the "try catch".

**Comment by ghost** [comment]
Let's include the android style into eclipse to get automatic formatting: http://stackoverflow.com/questions/7544769/coding-style-standards-for-android

File: src/com/fidesmo/sec/transceivedemo/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		
+		// Reference UI views and elements
+		final EditText uuidInput = (EditText) findViewById(R.id.uuid_input);		
+		Button transceiveToCardButton = (Button) findViewById(R.id.card_button); 
+		Button transceiveToDeviceFidelityButton = (Button) findViewById(R.id.devicefidelity_button); 
+		
+		// Link UI elements to listeners
+		transceiveToCardButton.setOnClickListener(new OnClickListener() {
+			@Override
+			public void onClick(View v) {
```
Comment (ghost) [inline src/com/fidesmo/sec/transceivedemo/MainActivity.java]: I think the content of the two OnClickListeners should be merged into one single function that takes the action string and uuid string as parameters. That will make very clear what is specific to how the sec-client is integrated.

File: src/com/fidesmo/sec/transceivedemo/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import android.widget.EditText;
+import android.widget.Toast;
+
+public class MainActivity extends Activity {
+	
+	// String for LogCat documentation
+	private final static String TAG = "Fidesmo-Transceive-Demo";
+	
+	// String constants defining the intent for using Fidesmo SEC Client
+	private final static String intentUri = "https://api.fidesmo.com/transaction/";
```
Comment (ghost) [inline src/com/fidesmo/sec/transceivedemo/MainActivity.java]: In Java C-style constants are used. Like `INTENT_URI`.

**Comment by ghost** [comment]
README.md should contain detailed build instructions.

**Comment by miguelcardo** [comment]
@petter-fidesmo Ready for second review!

**Comment by miguelcardo** [comment]
@petter-fidesmo Squashed.

**Pull request merged**